### ISSUE TITLE
[REVIEW]: Tyssue, en Epithelium simulation library

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2973]
**Submitting author:** @glyg (<a href="http://orcid.org/0000-0002-5444-5246">Guillaume Gay</a>)
**Repository:** <a href="https://github.com/damcb/tyssue" target ="_blank">https://github.com/damcb/tyssue</a>
**Version:** 0.9.0
**Editor:** @pdebuyl
**Reviewer:** @fcooper8472, @SergeDmi
**Archive:** <a href="https://doi.org/10.5281/zenodo.4817609" target="_blank">10.5281/zenodo.4817609</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/4645d8508a29875fa61c84ce2423654e/status.svg)](https://joss.theoj.org/papers/4645d8508a29875fa61c84ce2423654e)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/4645d8508a29875fa61c84ce2423654e"><img src="https://joss.theoj.org/papers/4645d8508a29875fa61c84ce2423654e/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/4645d8508a29875fa61c84ce2423654e/status.svg)](https://joss.theoj.org/papers/4645d8508a29875fa61c84ce2423654e)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/damcb/tyssue" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@fcooper8472 & @SergeDmi, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @pdebuyl know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/damcb/tyssue">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@glyg) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/damcb/tyssue">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@glyg) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @fcooper8472, @SergeDmi it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1101/2020.01.06.896266 is OK
- 10.1038/s41467-019-10720-0 is OK
- 10.1101/704932 is OK
- 10.1038/s41467-018-06497-3 is OK
- 10.1242/dev.146761 is OK
- 10.1098/rstb.2015.0520 is OK
- 10.1038/nphys3471 is OK
- 10.1038/nature14152 is OK
- 10.1007/s10237-015-0724-7 is OK
- 10.1007/s10237-012-0430-7 is OK
- 10.1016/J.CUB.2010.06.075 is OK
- 10.1016/J.CUB.2007.11.049 is OK
- 10.1016/S0021-9290(02)00006-4 is OK
- 10.1016/S0074-7696(08)62339-6 is OK
- 10.1016/0022-5193(78)90315-6 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by pdebuyl** [comment]
@fcooper8472 @SergeDmi make sure to accept the invitation to the reviewers group and to have a look at the reviewer guidelines linked to at the top of this review page.

The review process will happen in this issue page, so questions to the author or to me can be added as comments here.

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02973/joss.02973/10.21105.joss.02973.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02973/joss.02973/10.21105.joss.02973.pdf) :page_facing_up: :point_left:

**Comment by SergeDmi** [comment]
# Review
## Software structure and installation
I have actually little to say about the software. It is open source, well documented, easy to install with conda (although conda may have its pitfalls, it worked at first try for me). 

## Scholar effort
I think tyssue is a fantastic effort that really fills a need for well defined, reproducible physical models in the field, while allowing a lot of freedom for the user. It is clearly a fully-featured sofware.

## Software usage
Tyssue is an API, which allows a broad use, at the expense of simplicity for the non-programmer. My main concern is the lack of gradual difficulty in the tutorials : reproducing research results is great, but gently steering the user might be missing.
While example 5 is simple as claimed, example 2 is rather complex. To engage scientists to use the software, step-by-step tutorial, or a better gradation of difficulty in the notebook examples, seem required to smoothen the steep learning curve.

## Article
### Statement of need
The statement of need does a good job of explaining the context, but lacks in describing the actual need for the software. A mention of existing software is also lacking. Instead, this paragraph gives a short description of Tyssue.

### Software description
I think the article fails to convey the organization of Tyssue, that "Separates structure, geometry and models" (better explained in the readme). As a consequence, it is not clear that the models (referenced in the paragraph *mechanics*) can be chosen at will between Euler, Gradient descent, Quasi-static, and possibly other and user defined ; the same applies to *topology*. 

# Recommendation 
The software and article definitely should be published in JOSS. Usage examples should be streamlined. Article should highlight the need for this software as well as software organisation and capabilities. A link to the examples and doc should be provided.

**Comment by pdebuyl** [comment]
@SergeDmi thank you very much for the review.

@glyg you can choose to address directly @SergeDmi 's review or to wait for @fcooper8472 's review. Let us know how you wish to proceed.

**Comment by glyg** [comment]
Hi @pdebuyl and @SergeDmi 

Thank you a lot for your review Serge. @sophietheis and I are giving a workshop on tyssue next week, which will allow to test in the field a better tutorial.

If this is ok with you, we will answer Serge's remarks only after that (within the next 2 weeks).

**Comment by whedon** [comment]
:wave: @SergeDmi, please update us on how your review is going.

**Comment by pdebuyl** [comment]
@SergeDmi our bot is sometimes a bit zealous, sorry :-)

**Comment by pdebuyl** [comment]
Hi @glyg , any update?

**Comment by glyg** [comment]
Hi @pdebuyl we meet with Sophie on Friday to workout a revision plan, we gave a workshop last week that should help address Serge's remarks. I'll update on Monday

**Comment by glyg** [comment]
Hi @pdebuyl - The update to the doc is going on under this issue: https://github.com/DamCB/tyssue-demo/issues/4 - we are going back through the notebooks and trying to follow @SergeDmi 's advice to be more progressive. Also some newest notions as the `EventManager` haven't been documented properly yet. 

As for the paper itself, we thought it might be better to have the comments from the second reviewer before we undertake the revisions.

**Comment by pdebuyl** [comment]
@fcooper8472 can you address the "code" part of the review already (general checks, functionality, and documentation)?

**Comment by pdebuyl** [comment]
@glyg you can proceed along DamCB/tyssue-demo#4 . You can also address @SergeDmi 's paper comments as that won't intersect yet with the review of @fcooper8472

**Comment by glyg** [comment]
ok, I'll try to make some progress at the end of the week, then Sophie (who's absent this week) will relay

**Comment by pdebuyl** [comment]
@sophietheis ping

**Comment by sophietheis** [comment]
Hi, 
I am working on demo notebook, and I am also starting to improve the manuscript according to @SergeDmi comments. 
Modifications are on my remote, I will start a PR tomorrow.

**Comment by pdebuyl** [comment]
@fcooper8472 ping

**Comment by sophietheis** [comment]
Hi @SergeDmi 

You can find in the [PR](https://github.com/DamCB/tyssue/pull/225) the modification made to our manuscript according to your comments. 

1. Statement of need
_The statement of need does a good job of explaining the context, but lacks in describing the actual need for the software. A mention of existing software is also lacking. Instead, this paragraph gives a short description of Tyssue._

**We are agree that there is a lack of information about the need and the existing software. We have expanded this part with reference with several publish work :** 

Several vertex models have been developed in the past few years to describe the physics of epithelia (for a review, see [@Alt:2017]), and common features can be identified. Several kinds of models have already been published. The apical vertex model has been used several times to study topology changes during morphogenetic movement in _Drosophila, Hydra and Xenopus_([@Staple:2010], [@Farhadifar:2007], [@Aegerter:2012]). Associated with protein dynamics, it has been used to study the effect of protein position on tissue organisation in zebrafish retina ([@Salbreux:2012]). 3D vertex model have been used to study epithelium deformation due to normal development or to cancer development ([@Okuda:2015], [@Eritano:2020]).
Most of the time, models are developed for a specific biological question and are difficult to adapt to an other system, for severals reasons. With the `tyssue` library, we propose models which are adaptable and scalable with the fied research and the biological question. Topology and mechanics are implement independantly to improve the versatility of models.



2. Software description
_I think the article fails to convey the organization of Tyssue, that "Separates structure, geometry and models" (better explained in the readme). As a consequence, it is not clear that the models (referenced in the paragraph mechanics) can be chosen at will between Euler, Gradient descent, Quasi-static, and possibly other and user defined ; the same applies to topology._

**We add a scheme into the manuscript to highlight the modularity of our library ([new scheme](https://github.com/DamCB/tyssue/blob/master/doc/illus/Figure2.png)).**

**Comment by sophietheis** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.devcel.2021.01.005 is OK
- 10.2139/ssrn.3378007 is OK
- 10.1038/s41467-019-10720-0 is OK
- 10.1101/704932 is OK
- 10.1038/s41467-018-06497-3 is OK
- 10.1242/dev.146761 is OK
- 10.1098/rstb.2015.0520 is OK
- 10.1038/nphys3471 is OK
- 10.1038/nature14152 is OK
- 10.1007/s10237-015-0724-7 is OK
- 10.1371/journal.pcbi.1002970 is OK
- 10.1007/s10237-012-0430-7 is OK
- 10.1371/journal.pcbi.1002618 is OK
- 10.1242/dev.082800 is OK
- 10.1016/j.cell.2010.07.042 is OK
- 10.1016/J.CUB.2010.06.075 is OK
- 10.1016/J.CUB.2007.11.049 is OK
- 10.1016/S0021-9290(02)00006-4 is OK
- 10.1016/S0074-7696(08)62339-6 is OK
- 10.1016/0022-5193(78)90315-6 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by sophietheis** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02973/joss.02973/10.21105.joss.02973.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02973/joss.02973/10.21105.joss.02973.pdf) :page_facing_up: :point_left:

**Comment by fcooper8472** [comment]
My apologies for the late review: I had somehow managed to turn off notifications for this review, so had forgotten about it until Pierre emailed me.  Sorry!

---
I have now completed my review.

The software clearly passes the threshold scholarly effort.  It appears to be well engineered, easy to install, and very thoroughly documented including with extensive notebooks.

The paper itself is well written and explains the purpose and need of the software.

I recommend it for publication in JOSS.

---

The only minor comments I have are:

- Figure 1 (in particular) and Figure 2 (to a lesser extent) are pixelated in the paper. Is it possible to produce a higher-resolution version?
- The README contains the following, which doesn't seem to make chronological sense:
  ```
  Since version 0.3, this project is distributed under the terms of the General Public Licence.
  Versions 2.4 and earlier were distributed under the Mozilla Public Licence.
  ```
- There is a more recent publication for the Chaste software (https://doi.org/10.21105/joss.01848) which is now preferred for general citations, although I should state my conflict of interest as the first author on that paper

**Comment by glyg** [comment]
Thank you for the review @fcooper8472 , I'll fix the version problem, and we'll output higher resolution bitmaps for the figures, and fix the citation for Chaste

We are finishing a revamping of the documentation, it should be done by the end of the week, to fully account for @SergeDmi remarks, I'll comment here when it's done.

**Comment by pdebuyl** [comment]
Thank you for the review @fcooper8472 !

**Comment by pdebuyl** [comment]
gentle reminder @glyg @sophietheis

**Comment by glyg** [comment]
Hi @pdebuyl, sorry for the delay (all my fault), I'm polishing the doc tomorrow and deploying it (we switched from readthedocs to jupyter books)

**Comment by glyg** [comment]
Hi @pdebuyl @fcooper8472 and @SergeDmi - we updated the doc, now at [tyssue.io](http://tyssue.io), hopefully it is more detailed and progressive. 

We also fixed the citation od Chaste and the README issue

**Comment by pdebuyl** [comment]
Thanks @glyg for the update.

**Comment by pdebuyl** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02973/joss.02973/10.21105.joss.02973.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02973/joss.02973/10.21105.joss.02973.pdf) :page_facing_up: :point_left:

**Comment by kyleniemeyer** [comment]
Hi there, I'm the AEIC on duty this week, just looking to help submissions get past the finish line. It looks like @glyg and @sophietheis made some edits to the article in response to @SergeDmi's comments, so I'm thinking this may be nearly ready to accept?

I did notice some issues with references, that you can address:
- the "Cell Flow Reorients..." citation seems to be missing authors
- The Honda (1983) citation is missing the journal
- the Martin et al. citation is missing year and journal

**Comment by glyg** [comment]
Thanks @kyleniemeyer , checking those now

**Comment by pdebuyl** [comment]
@SergeDmi would you give a look at the updates?

**Comment by glyg** [comment]
Hi @pdebuyl , no news from @SergeDmi ? I'm not sure he reads his notifications

Best
Guillaume

**Comment by pdebuyl** [comment]
Hi @glyg you are right. I sent an email.

**Comment by SergeDmi** [comment]
Sorry for the late reply, I had notifications off indeed. 

All my comments have been addressed and I recommend the article for publication.

(there might be an issue with acknowledgements though, see :)
![Screenshot_20210521_085653](https://user-images.githubusercontent.com/20117767/119095515-aa0ba680-ba12-11eb-9edd-444ee2ee46b1.png)

**Comment by glyg** [comment]
thanks @SergeDmi ! We fixed the contributors links

**Comment by pdebuyl** [comment]
@whedon generate pdf

**Comment by pdebuyl** [comment]
Thank you @SergeDmi for completing the review!

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02973/joss.02973/10.21105.joss.02973.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02973/joss.02973/10.21105.joss.02973.pdf) :page_facing_up: :point_left:

**Comment by glyg** [comment]
Some links to notebooks were broken as we changed the organization of the doc, we're fixing them now

**Comment by sophietheis** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02973/joss.02973/10.21105.joss.02973.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02973/joss.02973/10.21105.joss.02973.pdf) :page_facing_up: :point_left:

**Comment by pdebuyl** [comment]
Hi, @glyg @sophietheis I'll proceed to the next step https://joss.readthedocs.io/en/latest/editing.html#after-reviewers-recommend-acceptance

I found the following grammatical and spelling errors on the first page. Can you fix them and run a spell checker on the article?

- "It's main focus" -> "Its main focus"
- "epihelia": is it "epithelia" ? (added "t").
- First sentence in statement of need is a list without end. The last entry should start with a "and" or an "or", or end with etc.
- "helps us understand" -> "helps us to understand"
- "how tissue" -> "how tissues"
- "to an other system" -> "to another system" or "to other systems"
- "the fiel research" -> "the field of research"
- "are implement" -> "are implemented"
- "independantly" -> "independently"

Also, you should combine citations with a ";" inside square brackets. See the [citation syntax](https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax).

You mention "the scientific Python tools" in the acknowledgments. Some of the core packages, among which NumPy, have citation guidelines: https://scipy.org/citing.html#numpy

Can you check the version number of tyssue associated with the submission, archive the code on zenodo (with correct metadata for the title and author list), and give an update here with the DOI?

**Comment by glyg** [comment]
Thanks @pdebuyl we'll go through all that. I'll make a release of the code & check the zenodo metadata et al. 
@sophietheis looks into the spelling and citations syntax.

We'll update you here (not today for me, I don't have enough time ^^)

**Comment by glyg** [comment]
Hi @pdebuyl I just made a release of tyssue, tagged [0.9.0](https://github.com/DamCB/tyssue/releases/tag/0.9.0)  with the DOI [10.5281/zenodo.4817609](https://doi.org/10.5281/zenodo.4817609) and @sophietheis addressed the spelling  citation issues in [issue 242](https://github.com/DamCB/tyssue/pull/242).

**Comment by pdebuyl** [comment]
@whedon set 0.9.0 as version

**Comment by whedon** [comment]
OK. 0.9.0 is the version.

**Comment by pdebuyl** [comment]
@whedon set 10.5281/zenodo.4817609 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4817609" target="_blank">10.5281/zenodo.4817609</a> is the archive.

**Comment by pdebuyl** [comment]
@glyg can you update the metadata of the zenodo archive (you can "edit" it via the "upload" page) to match those of the paper (title, etc)? I see that the author listing does not match, can you address that?

Depending on the situation, some names can be listed as "contributors" and not "authors" depending on their contribution. "gitter-badger" should be removed. Check our authorship guidelines: https://joss.readthedocs.io/en/latest/submitting.html#authorship

**Comment by glyg** [comment]
Thanks @pdebuyl - I fixed the title & author list, and added the ERC grant (that's neatly done !)

**Comment by pdebuyl** [comment]
Thank you for the changes. Can you remove html formatting in the md file?

There are still a few minor spelling errors, I'll try to go over the paper carefully in the coming days.

**Comment by glyg** [comment]
I removed the tags: 
https://github.com/DamCB/tyssue/commit/4d8f8624b5d94649f0cd3e0be3a83def3d9f711f

I'll try to read it again also. 
Thanks

**Comment by glyg** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02973/joss.02973/10.21105.joss.02973.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02973/joss.02973/10.21105.joss.02973.pdf) :page_facing_up: :point_left:

**Comment by glyg** [comment]
Fixed parentheses & some typos

@whedon generate pdf

**Comment by sophietheis** [comment]
> Fixed parentheses & some typos
> 
> @whedon generate pdf

Also fixed missing references.

**Comment by pdebuyl** [comment]
Hi @glyg and @sophietheis . I opened PR https://github.com/DamCB/tyssue/pull/244 with the remaining syntax issues. I added a section title for the "not statement of need" part of the paper and fix a minor grammatical error.

Can you have a look at it and let me know if it is suitable for you? A final request: in the acknowledgments, you include one of the co-authors. Can you change it to "We wish to thank the team of Magali Suzanne (...)"? The current situation is a bit weird.

Apart from these issues, I am good for approving the paper.

**Comment by glyg** [comment]
Thanks a lot @pdebuyl - I thought I did remove the tags [here](https://github.com/DamCB/tyssue/commit/4d8f8624b5d94649f0cd3e0be3a83def3d9f711f#diff-0403ef06adf405f7b310b4518bd6a3559854f54c61676f676ce9cbfee7172ab6) but I guess there was some merge problem, sorry you had to do it again. 

>  The current situation is a bit weird.

Ha yes good point, I saw it and was not sure how to get around it. I rephrased it now.

**Comment by pdebuyl** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.devcel.2021.01.005 is OK
- 10.2139/ssrn.3378007 is OK
- 10.21105/joss.01848 is OK
- 10.1038/s41592-019-0686-2 is OK
- 10.1038/s41586-020-2649-2 is OK
- 10.1038/s41467-019-10720-0 is OK
- 10.1101/704932 is OK
- 10.1038/s41467-018-06497-3 is OK
- 10.1242/dev.146761 is OK
- 10.1098/rstb.2015.0520 is OK
- 10.1038/nphys3471 is OK
- 10.1038/nature14152 is OK
- 10.1007/s10237-015-0724-7 is OK
- 10.1007/s10237-012-0430-7 is OK
- 10.1371/journal.pcbi.1002618 is OK
- 10.1242/dev.082800 is OK
- 10.1016/j.cell.2010.07.042 is OK
- 10.1016/J.CUB.2010.06.075 is OK
- 10.25080/Majora-92bf1922-00a is OK
- 10.1016/J.CUB.2007.11.049 is OK
- 10.1109/MCSE.2007.55 is OK
- 10.1016/S0021-9290(02)00006-4 is OK
- 10.1016/S0074-7696(08)62339-6 is OK
- 10.1016/0022-5193(78)90315-6 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by pdebuyl** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02973/joss.02973/10.21105.joss.02973.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02973/joss.02973/10.21105.joss.02973.pdf) :page_facing_up: :point_left:

**Comment by pdebuyl** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.devcel.2021.01.005 is OK
- 10.2139/ssrn.3378007 is OK
- 10.21105/joss.01848 is OK
- 10.1038/s41592-019-0686-2 is OK
- 10.1038/s41586-020-2649-2 is OK
- 10.1038/s41467-019-10720-0 is OK
- 10.1101/704932 is OK
- 10.1038/s41467-018-06497-3 is OK
- 10.1242/dev.146761 is OK
- 10.1098/rstb.2015.0520 is OK
- 10.1038/nphys3471 is OK
- 10.1038/nature14152 is OK
- 10.1007/s10237-015-0724-7 is OK
- 10.1007/s10237-012-0430-7 is OK
- 10.1371/journal.pcbi.1002618 is OK
- 10.1242/dev.082800 is OK
- 10.1016/j.cell.2010.07.042 is OK
- 10.1016/J.CUB.2010.06.075 is OK
- 10.25080/Majora-92bf1922-00a is OK
- 10.1016/J.CUB.2007.11.049 is OK
- 10.1109/MCSE.2007.55 is OK
- 10.1016/S0021-9290(02)00006-4 is OK
- 10.1016/S0074-7696(08)62339-6 is OK
- 10.1016/0022-5193(78)90315-6 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2365

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2365, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by pdebuyl** [comment]
Thanks all for the submission and review! The editor-in-chief in rotation will pick up the process soon for finalizing the publication.

**Comment by glyg** [comment]
Great!!! Thanks a lot :)

**Comment by kthyng** [comment]
@glyg I see a small fix in your paper to make: in the parenthetical statement "for a review, see", the reference there should be in-line, not parenthetical. So, you should remove the [] around that reference.

**Comment by glyg** [comment]
Thanks @kthyng good catch, I fixed it

**Comment by glyg** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02973/joss.02973/10.21105.joss.02973.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02973/joss.02973/10.21105.joss.02973.pdf) :page_facing_up: :point_left:

**Comment by kthyng** [comment]
Ok I think everything looks good!

**Comment by kthyng** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1401958263667691520) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2369
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.02973](https://doi.org/10.21105/joss.02973)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by kthyng** [comment]
Congrats on your new publication @glyg! Thanks to @pdebuyl for editing and to reviewers @fcooper8472 and @SergeDmi for your time, hard work, and expertise!!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02973/status.svg)](https://doi.org/10.21105/joss.02973)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02973">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02973/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02973/status.svg
   :target: https://doi.org/10.21105/joss.02973
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02973/status.svg)](https://doi.org/10.21105/joss.02973)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by glyg** [comment]
this is really great!!! Thank you all for the great review process!!!
